qdrant-landing/content/articles/what-is-quantization.md

```diff
@@ -0,0 +1,249 @@
+
+If your dataset has **millions of vectors**, the memory and processing demands are very significant.
+
+To understand why this process is so computational demanding, let's take a look at the nature of the [HNSW index.](https://qdrant.tech/documentation/concepts/indexing/#vector-index)
+
+The HNSW (Hierarchical Navigable Small World) index organizes vectors in a layered graph, connecting each vector to its nearest neighbors. At each layer, the algorithm narrows down the search area until it reaches the lower layers, where it efficiently finds the closest matches to the query.
+
+<img src="/articles_data/what-is-vector-quantization/hnsw-search.png" alt="HNSW Search visualization" width="300">
+
+So each time a new vector is added, the system must determine its position in the existing graph, a process similar to searching. This makes both inserting and searching for vectors complex operations.
+
+One of the key challenges with the HNSW index is that it requires a lot of **random reads** and **sequential traversals** through the graph. This makes the process computationally expensive, especially when you're dealing with millions of high-dimensional vectors. 
+
+The system has to jump between various points in the graph in an unpredictable way. This unpredictability makes it hard to optimize, and as the dataset grows, the memory and processing requirements increase significantly.
+
+
+<img src="/articles_data/what-is-vector-quantization/hnsw-search.png" alt="HNSW Search visualization" width="300">
+
+[to add gif]
+
+And because vectors need to be stored in **fast storage** like **RAM** or **SSD** for low-latency searches, as the size of the data grows, so does the cost of storing and processing it efficiently.
+
+Quantization offers a solution by compressing vectors to smaller memory sizes, making the process more efficient. 
+
+There are several methods to achieve this, and here we will focus on three main ones:
+
+
+![Types of Quantization: 1. Scalar Quantization, 2. Product Quantization, 3. Binary Quantization](/articles_data/what-is-vector-quantization/types-of-quantization.png)
+
+
+## 1. What is Scalar Quantization?
+
+![](/articles_data/what-is-vector-quantization/astronaut-mars.jpg)
+
+In Qdrant, each dimension is represented by a float32 value, which uses **4 bytes** of memory. When using [Scalar Quantization](https://qdrant.tech/documentation/guides/quantization/#scalar-quantization), we are mapping our vectors to a range that the smaller int8 type can represent. An int8 can store 256 values (from -128 to 127, or 0 to 255) which uses only **1 byte.** This results in a **75% reduction** in memory size.
+
+For example, if our data lies in the identified range of -1.0 to 1.0, Scalar Quantization will transform these values to a range that int8 can represent, that is, within -128 to 127. So, the system **maps** the float32 values into this range.
```

Comment:
I checked SQ article https://qdrant.tech/articles/scalar-quantization/
And I see a common mistake. In SQ we use range `[0..127]`. Yeah, we skip one bit from a byte in SQ. We did that to achieve better SIMD acceleration. I don't know what to do: fix in both places or just say "nevermind" =)

Reply:
Is there a specific reason why that is? Wouldn't the SIMD still work with the full range of an int8 (from -128 to 127)?